### ISSUE TITLE
proxychains Set multi-level agen,tor as a proxyheader has problems

### ISSUE THREAD

**Issue by cbedhoam620** [rofl0r/proxychains-ng#489]
```properties
socks5 127.0.0.1 9050  # Even the socks port of Tor on the public web cannot be replaced here
socks5 PROXY2 1080
socks5 PROXY3 1080
```
I use `proxychains -f /etc/Torproxylist.conf curl ident.me`, happen "socket error or timeout".

But it doesn't matter if I use any of the proxies in the proxy chain alone, or if I comment out Tor's proxies, the rest of the proxies can still form the proxy chain

**Comment by rofl0r** [comment]
in order to use tor proxy on 127.0.0.1 you have to run tor service on your machine.

**Comment by cbedhoam620** [comment]
> 为了在 127.0.0.1 上使用 tor 代理，你必须在你的机器上运行 tor 服务。

Yes, I have enabled the Tor service on this machine, but what I mean is that no matter whether you fill in 127.0.0.1 or the public network address of this machine, it will not take effect. But it is possible to use Tor alone as a proxy, but it exists in the proxy list Multiple agents will not take effect

**Comment by rofl0r** [comment]
with "multiple agents" you mean multiple proxy ip addresses ?
you need to use a chain type like strict, and the ip addresses there need to be accessible from the exit node your tor circuit uses (and they need to be generally usable/online of course)
